syntax = "proto3";

package silenteight.datasource.api.name;

import "google/api/annotations.proto";
import "silenteight/datasource/api/common/common.proto";

option java_package = "com.silenteight.datasource.api.name";
option java_outer_classname = "NameInputProto";
option java_multiple_files = true;

/**
  Intended to be used by
  - name agent
  - guess gender agent
  - region agent
 */
service NameInputService {
  rpc BatchGetMatchNameInputsV1 (silenteight.datasource.api.common.BatchGetMatchAgentInputsRequest) returns (stream BatchGetMatchNameInputsV1Response) {
    option (google.api.http) = {
      post: "/agent-inputs/name/versions/1/matches:batchGet",
      body: "*"
    };
  }
}

message BatchGetMatchNameInputsV1Response {
  repeated NameInput name_inputs = 1;
}

message NameInput {
  string match = 1;
  repeated NameFeatureInput name_feature_inputs = 2;
}

message NameFeatureInput {
  string feature = 1;
  repeated AlertedPartyName alerted_party_names = 2;
  repeated WatchlistName watchlist_names = 3;
  EntityType alerted_party_type = 4;
  // List of strings that screening engine matched (used only for Organization)
  repeated string matching_texts = 5;

  enum EntityType {
    INDIVIDUAL = 0;
    ORGANIZATION = 1;
  }
}

message AlertedPartyName {
  string name = 1;
}

message WatchlistName {
  string name = 1;
  WatchlistNameType type = 2;

  enum WatchlistNameType {
    NAME = 0;
    ALIAS = 1;
  }
}
